x/incentives/keeper/grpc_query.go

```diff
@@ -86,6 +97,36 @@ func (k Keeper) ActiveGauges(goCtx context.Context, req *types.ActiveGaugesReque
 	return &types.ActiveGaugesResponse{Data: gauges, Pagination: pageRes}, nil
 }
 
+// ActiveGaugesPerDenom returns active gauges for the specified denom
+func (k Keeper) ActiveGaugesPerDenom(goCtx context.Context, req *types.ActiveGaugesPerDenomRequest) (*types.ActiveGaugesPerDenomResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	gauges := []types.Gauge{}
+	store := ctx.KVStore(k.storeKey)
+	valStore := prefix.NewStore(store, types.KeyPrefixActiveGauges)
+
+	pageRes, err := query.FilteredPaginate(valStore, req.Pagination, func(key []byte, value []byte, accumulate bool) (bool, error) {
+		guageIDs := k.getAllGaugeIDsByDenom(ctx, req.Denom)
+		for _, id := range guageIDs {
+			gauge, err := k.GetGaugeByID(ctx, id)
+			if err != nil {
+				panic(err)
+			}
+			if gauge.IsPerpetual {
+				gauges = append(gauges, *gauge)
+			} else if gauge.NumEpochsPaidOver < gauge.FilledEpochs {
+				gauges = append(gauges, *gauge)
+			}
```

Comment:
Is there no direct keeper method to determine if the gauge is active?

Reply:
Nope there isn't

Reply:
Which, I think would be better to make instead of making this part of the code dangle in `grpc_query.go`. Let me methodize it and move it to keeper

Reply:
There actually is, let me replace this to the method

Reply:
Done in https://github.com/osmosis-labs/osmosis/pull/572/commits/cbabd871a117231d1e3db4a0257d46d555c098de.